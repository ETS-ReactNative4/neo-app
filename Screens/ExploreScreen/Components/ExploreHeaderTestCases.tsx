import React from "react";
import { ITestCase } from "../../../TypeInterfaces/TestCases/ITestCases";
import ExploreHeader from "./ExploreHeader";

const ExploreHeaderTestCases: ITestCase[] = [
  {
    title: "Explore Header",
    Component: <ExploreHeader />
  }
];

export default ExploreHeaderTestCases;
