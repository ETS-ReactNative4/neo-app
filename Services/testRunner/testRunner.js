import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
