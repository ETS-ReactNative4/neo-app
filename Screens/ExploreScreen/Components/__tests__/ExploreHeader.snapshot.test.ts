import ExploreHeaderTestCases from "../ExploreHeaderTestCases";
import testRunner from "../../../../Services/testRunner/testRunner";

ExploreHeaderTestCases.map(testRunner);
