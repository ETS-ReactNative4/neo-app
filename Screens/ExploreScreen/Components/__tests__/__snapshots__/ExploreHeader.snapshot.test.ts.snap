// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore Header 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(255, 255, 255)",
        "borderBottomColor": "rgba(239, 239, 239, 1)",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.8}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 56,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "scaleX": -1,
          },
        ],
        "width": 56,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(83, 87, 109, 1)",
            "fontSize": 16,
          },
          undefined,
          Object {
            "fontFamily": "voyager",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      Ó•ê
    </Text>
  </TouchableOpacity>
  <Image
    onError={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://pyt-images.imgix.net/images/web_app/logo/pickyourtrail-logo.png",
      }
    }
    style={
      Object {
        "height": 20,
        "width": 112,
      }
    }
  />
</View>
`;
